frontend/lib/activation/deeplink_activation.dart

```diff
@@ -0,0 +1,211 @@
+                                        }
+                                      : null,
+                                  child: Text(t.activation.buttonText),
+                                )
+                              ],
+                          },
+                        ],
+                      ),
+                    ),
+                  ],
+                )),
+          );
+        });
+  }
+}
+
+class _WarningText extends StatelessWidget {
+  final String text;
+
+  const _WarningText(this.text);
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Icon(Icons.warning, color: Theme.of(context).colorScheme.secondary),
+        Text(text, textAlign: TextAlign.center)
+      ],
+    );
+  }
+}
+
+DynamicActivationCode? getActivationCode(BuildContext context, String base64qrcode) {
+  try {
+    if (base64qrcode.length % 4 == 0) {
+      final activationCode =
+          const ActivationCodeParser().parseQrCodeContent(const Base64Decoder().convert(base64qrcode));
+      return activationCode;
+    }
+    return null;
+  } on QrCodeParseException catch (e, _) {
+    debugPrint('Der eingescannte Code ist ungültig.');
+    return null;
+  }
+}
+
+Future<void> activateCard(BuildContext context, DynamicActivationCode activationCode) async {
+  final messengerState = ScaffoldMessenger.of(context);
+  final userCodesModel = Provider.of<UserCodeModel>(context, listen: false);
+  final client = GraphQLProvider.of(context).value;
+  final projectId = Configuration.of(context).projectId;
+
+  final activationResult = await activateCode(
+    client: client,
+    projectId: projectId,
+    activationSecretBase64: const Base64Encoder().convert(activationCode.activationSecret),
+    cardInfoHashBase64: activationCode.info.hash(activationCode.pepper),
+    overwriteExisting: true,
+  );
+
+  switch (activationResult.activationState) {
+    case ActivationState.success:
+      if (activationResult.totpSecret == null) {
+        throw const ActivationInvalidTotpSecretException();
+      }
+      final totpSecret = const Base64Decoder().convert(activationResult.totpSecret!);
+      final userCode = DynamicUserCode()
+        ..info = activationCode.info
+        ..pepper = activationCode.pepper
+        ..totpSecret = totpSecret;
+      userCodesModel.insertCode(userCode);
+      GoRouter.of(context).pushReplacement('$homeRouteName/$identityTabIndex');
+      messengerState.showSnackBar(
+        SnackBar(
+          backgroundColor: Theme.of(context).colorScheme.primary,
+          content: Text(t.activation.activationSuccessful),
+        ),
+      );
+      break;
+    case ActivationState.failed:
+      await QrParsingErrorDialog.showErrorDialog(
+        context,
+        t.activation.invalidCode,
+      );
+      break;
+    case ActivationState.didNotOverwriteExisting:
+      throw const ActivationDidNotOverwriteExisting();
```

Comment:
I am unsure how the error handling is done here. Can this case ever happen?
If not, we should probably add a comment as to why this should never happen. If it could happen, then shouldn't we rather show an error dialog to the user?

Reply:
extract function to `utils` and reuse in `ActivationCodeScannerPage` and `DeepLinkActivation`